java/ant.grammar/src/org/netbeans/modules/ant/grammar/AntHighlightsLayerFactory.java

```diff
@@ -38,18 +34,16 @@
 public class AntHighlightsLayerFactory implements HighlightsLayerFactory {
 
     public @Override HighlightsLayer[] createLayers(final Context context) {
-        AttributeSet _attrs = MimeLookup.getLookup("text/x-jsp").lookup(FontColorSettings.class).getTokenFontColors("expression-language"); // NOI18N
-        final AttributeSet attrs;
-        if (_attrs == null) {
-            // Fallback from web.core.syntax/src/org/netbeans/modules/web/core/syntax/resources/fontsColors.xml:
-            SimpleAttributeSet _sattrs = new SimpleAttributeSet();
-            _sattrs.addAttribute(StyleConstants.Background, new Color(0xe3f2e1));
-            attrs = _sattrs;
-        } else {
-            attrs = _attrs;
-        }
-        return new HighlightsLayer[] {HighlightsLayer.create("ant", ZOrder.SYNTAX_RACK.forPosition(10), true,
-                new AntHighlightsContainer((AbstractDocument) context.getDocument(), attrs))}; // NOI18N
+        HighlightsLayer highlighter = HighlightsLayer.create(
+            "ant",  // NOI18N
+            ZOrder.SYNTAX_RACK.forPosition(10), 
+            true, 
+            new AntHighlightsContainer(
+                (AbstractDocument) context.getDocument(),
+                MimeLookup.getLookup("text/x-jsp").lookup(FontColorSettings.class).getTokenFontColors("expression-language") // NOI18N
```

Comment:
the fact that this uses the JSP EL setting for ant means that it isn't there if the module isn't loaded. My NB often disables some enterprise modules on startup due to ergonomics, in that case it loses its color settings until the module is activated again.

Reply:
right. It will use the base colors if the theme does not provide a color for it. FlatLaf light has a highlighting layer defined for those property values, the dark variant has not, so it will look like on the screenshot. Which is better than using the hardcoded value which was meant for light themes. This is especially a problem while trying to create your own theme.

thanks for reviewing!